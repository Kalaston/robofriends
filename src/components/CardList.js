import React from "react";
import Card from "./Card";

export default props => {
  const cardList = props.robots.map(robot => (
    <Card id={robot.id} name={robot.name} email={robot.email} key={robot.id} />
  ));
  return cardList;
};
