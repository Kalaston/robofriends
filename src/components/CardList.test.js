import React from 'react';
import { shallow } from 'enzyme';
import CardList from './CardList';
// import Enzyme from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
// Enzyme.configure({ adapter: new Adapter() });

it('expect to render Card Component', () => {
  const mockRobots = [
    {
      id: 1,
      name: 'Juan Snow',
      username: 'JuanNieves',
      email: 'juan@gmail.com'
    }
  ];
  expect(shallow(<CardList robots={mockRobots} />)).toMatchSnapshot();
});
