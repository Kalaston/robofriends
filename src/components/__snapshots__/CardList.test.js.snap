// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card Component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    robots={
        Array [
            Object {
              "email": "juan@gmail.com",
              "id": 1,
              "name": "Juan Snow",
              "username": "JuanNieves",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "1",
    "nodeType": "function",
    "props": Object {
      "email": "juan@gmail.com",
      "id": 1,
      "name": "Juan Snow",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "1",
      "nodeType": "function",
      "props": Object {
        "email": "juan@gmail.com",
        "id": 1,
        "name": "Juan Snow",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
